CMakeLists.txt

```diff
@@ -133,6 +133,9 @@ if (FLANN_FOUND)
     include_directories(SYSTEM "${FLANN_INCLUDE_DIRS}")
 endif()
 
+find_package( Eigen REQUIRED )
+include_directories(SYSTEM "${Eigen_INCLUDE_DIRS}")
+
```

Comment:
I will try to make the Eigen dependency optional post-merge. Eigen is a relatively safe choice for performing linear algebra operations. Since it's header-only and widely available through package managers, I don't think the dependency is a problem.